src/textual/widgets/_data_table.py

```diff
@@ -1125,6 +1125,8 @@ def _update_column_widths(self, updated_cells: set[CellKey]) -> None:
             else:
                 column.content_width = max(new_content_width, label_width)
 
+        self._require_update_dimensions = True
```

Comment:
We're updating the width of a column, so we need to set this flag to ensure the virtual size gets updated next time we go on_idle.